openfl/federated/data/sources/verifiable_dataset_info.py

```diff
@@ -0,0 +1,167 @@
+# Copyright 2020-2025 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+"""This module contains the VerifiableDatasetInfo class."""
+
+import json
+from enum import Enum
+from hashlib import sha384
+from pathlib import Path
+from typing import List
+
+from openfl.federated.data.sources.data_source import DataSource, DataSourceType
+from openfl.federated.data.sources.local_data_source import LocalDataSource
+
+
+class DatasetFormat(Enum):
+    """Enum for the different dataset commitment formats."""
+
+    VERBOSE = "verbose_dataset"
+    CONCISE = "concise_dataset"
+
+
+def filter_non_serializable(obj):
+    """Filter out methods and non-serializable objects."""
+    serializable_dict = {}
+    for key, val in obj.__dict__.items():
+        if callable(val):
+            continue  # Skip methods
+        if isinstance(val, Enum):
+            val = val.value  # Convert Enum to its value
+        elif isinstance(val, Path):
+            val = str(val)  # Convert Path to string
+        serializable_dict[key] = val
+    return serializable_dict
+
+
+class VerifiableDatasetInfo:
```

Comment:
Why not just `VerifiableDataset` ?

Reply:
Because it is not really a dataset, we can't use it for loading the data.